File: problems/3345-smallest-divisible-digit-product-i/solution.c
int smallestNumber(int n, int t) {
    while (1) {
        int product = 1;
        int temp = n;
        while (temp > 0) {
            product *= temp % 10;
            temp /= 10;
        }
        if (product % t == 0) {
            return n;
        }
        n++;
    }
}

File: problems/3345-smallest-divisible-digit-product-i/solution.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        while True:
            product = 1
            temp = n
            while temp > 0:
                digit = temp % 10
                product *= digit
                temp //= 10
            
            if t == 1 or (product % t == 0 and t > 0) or (t == 0 and product == 0):
                return n
            
            n += 1